swarm/api/routes/validation_utils.py

```
@@ -0,0 +1,39 @@
+    """
+    try:
+        return validate_path_component(value, name)
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "invalid_path_component",
+                "message": str(e),
+                "details": {name: value},
+            },
+        )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)